qchem/pennylane_qchem/qchem/structure.py

```diff
@@ -248,7 +256,10 @@ def meanfield(
 
     Args:
         name (str): molecule label
```

Comment:
I'm still confused about having to pass `name` as an argument. In my mind, it should be possible for this function to work without having to create any local files. Am I wrong? Is this an issue related to Psi4 and PySCF?

Reply:
PySCF/Psi4 creates an HDF5-format file from which the HF quantities required to build H are retrieved. We offer the user the possibility to label the molecule in order not to overwrite this file for different molecules. 
For example, if you want to perform quantum simulations of different approximations to the Hamiltonian (e.g., different active orbitals) of a given molecule you want to make sure you are reading the right mean field input:

```
hf_file = "./pyscf/6-31g/h2o.hdf5"
H = qchem.convert_observable(qchem.decompose(hf_file, mapping="jordan_wigner", core=... , active=...))
```  
so you don't need to run the HF again. If we think this feature is mainly useful for advanced users we could make ``name`` a keyword argument in ``meanfield()`` and ``molecular_hamiltonian()``.

```
def meanfield(symbols, coordinates, name="mol", charge=0, mult=1, basis="sto-3g", package="pyscf", outpath=".") 
```
what do you think?. I like it.

Reply:
I like it too! 

But to be clear, Psi4 and PySCF always create these local files? This is not something that can be avoided?

Reply:
I don't think so because OpenFermion uses this file to load the quantities we need to build Hamiltonian.

Reply:
Is there a way to 'skip' this, though? OpenFermion does not allow you to build a Hamiltonian using Python objects outputted by Psi4/pyscf?

Reply:
OpenFermion-Psi4/Pyscf plugins add python scripts to the input files of Psi4/pyscf to fish the electronic integrals and some other things and output them to the HDF5-format file which is used to populate the attributes of the ``MolecularData`` class (https://github.com/quantumlib/OpenFermion/blob/master/src/openfermion/chem/molecular_data.py)